bindings/wasm/examples/src/explorer-did.js

```diff
@@ -0,0 +1,243 @@
+    const config = Config.fromNetwork(CLIENT_CONFIG.network);
+
+    // Create a client instance to publish messages to the Tangle.
+    const client = Client.fromConfig(config);
+
+    // Generate a new ed25519 public/private key pair.
+    const key = new KeyPair(KeyType.Ed25519);
+
+    // Create a DID Document (an identity) from the generated key pair.
+    const doc = new Document(key, CLIENT_CONFIG.network.toString());
+
+    // Sign the DID Document with the generated key.
+    doc.signSelf(key, doc.defaultSigningMethod().id.toString());
+
+    // Publish the Identity to the IOTA Network, this may take a few seconds to complete Proof-of-Work.
+    const receipt = await client.publishDocument(doc);
+
+    console.log(doc.id.toString());
+
+
+    // ===========================================================================
+    // Integration Chain Update 1
+    // ===========================================================================
+
+    console.log("start integration update 1");
+
+    // Prepare an integration chain update, which writes the full updated DID document to the Tangle.
+    const intDoc1 = Document.fromJSON(doc.toJSON()); // clone the Document
+
+    // Add a new VerificationMethod with a new KeyPair, with the tag "keys-1"
+    const keys1 = new KeyPair(KeyType.Ed25519);
+    const method1 = VerificationMethod.fromDID(intDoc1.id, keys1, "keys-1");
+    intDoc1.insertMethod(method1, MethodScope.VerificationMethod());
+
+    // Add the `messageId` of the previous message in the chain.
+    // This is REQUIRED in order for the messages to form a chain.
+    // Skipping / forgetting this will render the publication useless.
+    intDoc1.metadataPreviousMessageId = receipt.messageId;
+    intDoc1.metadataUpdated = Timestamp.nowUTC();
+
+    // Sign the DID Document with the original private key.
+    intDoc1.signSelf(key, intDoc1.defaultSigningMethod().id.toString());
+
+    // Publish the updated DID Document to the Tangle, updating the integration chain.
+    // This may take a few seconds to complete proof-of-work.
+    const intReceipt1 = await client.publishDocument(intDoc1);
+
+
+    // ===========================================================================
+    // Spam on first integration update.
+    // ===========================================================================
+
+    const intIndex = intDoc1.integrationIndex();
+    await client.publishJSON(intIndex, { "intSpam:1": true });
+    await client.publishJSON(intIndex, { "intSpam:2": true });
+
+    // ===========================================================================
+    // Integration Chain Update 2
+    // ===========================================================================
+
+    console.log("start integration update 2");
+
+    // Prepare an integration chain update, which writes the full updated DID document to the Tangle.
+    const intDoc2 = Document.fromJSON(intDoc1.toJSON()); // clone the Document
```

Comment:
I wonder if we need to expose a dedicated `clone` function in each binding struct for easy deep-clones...